content/terraform/v1.12.x/docs/language/meta-arguments.mdx

```diff
@@ -0,0 +1,211 @@
+
+### Guidance
+
+- [Providers](/terraform/language/providers)
+- [Select an alternate provider configuration](/terraform/language/data-sources#alternate-provider-configurations) when querying a data source
+
+### References
+
+- [`provider` argument in `check` blocks](/terraform/language/block/check#provider)
+- [`provider` argument in `data` blocks](/terraform/language/block/data#provider) 
+- [`provider` argument in `ephemeral` blocks](/terraform/language/block/ephemeral#provider)
+- [`provider` argument in `import` block ](/terraform/language/block/import#provider)
+- [`provider` argument in `resource` blocks](/terraform/language/block/resource#provider) 
+
+## `providers`
+
+By default, child modules inherit the default provider configurations of their parent module, but you can specify an alternate provider configuration in the `module` block. The `providers` argument instructs Terraform to use the reference provider configuration to create the module resources. 
+
+Some modules may contain resources that require different providers. As a result, you can specify alternate configurations for each provider supported by the module. Refer to the module documentation for details about the providers it supports. 
+
+When you add a `providers` argument, child modules only have access to the provider configurations you specify. Use the `providers` argument in the following scenarios:
+
+- To use an alternate provider configuration for a child module.
+- To configure a module that requires multiple configurations of the same provider.
+
+### Examples
+
+- [Apply different provider configurations](/terraform/language/block/module#apply-different-provider-configurations-for-module-resources)  for module resources
+
+### Guidance
+
+- [Providers](/terraform/language/providers)
+
+### Reference
+
+- [`providers` argument in `module` blocks](/terraform/language/block/module#providers)
+- [Providers module development documentation](/terraform/language/modules/develop/providers) 
+
+## `lifecycle`
+
+Terraform performs the following operations when you apply a configuration: 
+
+1. Creates resources that exist in the configuration but are not associated with a real infrastructure object in the state.
+1. Destroys resources that exist in the state but no longer exist in the configuration.
+1. Updates in-place resources whose arguments have changed.
+1. Destroys and re-create resources whose arguments have changed but that cannot be updated in-place due to remote API limitations.
+
+The `lifecycle` block accepts a rule that customizes how Terraform performs the lifecycle stages for each resource. Support for each `lifecycle` rule varies across Terraform configuration blocks. Refer to the reference documentation for the Terraform block you are adding to your configuration for details. The following table describes support for each `lifecycle` rule:
+
+You can apply the following rules to all `resource` blocks:
```

Comment:
I have a feeling you meant to remove this and use the sentence above as a table intro
```suggestion
```